prov/shm/src/smr_util.c

```diff
@@ -554,8 +555,7 @@ int smr_map_add(const struct fi_provider *prov, struct smr_map *map,
 	if (ret) {
 		assert(ret == -FI_EALREADY);
 		*id = (intptr_t) node->data;
-		ofi_spin_unlock(&map->lock);
```

Comment:
This seems to change the behavior, you have `smr_map_to_region` in the `out` while before your change it seems to return without map, is it expected? @aingerson

Reply:
I think I understand what is going on. Its trying to map to region when it shouldnt because the region already exists... so either I need to revert and return 0 or check if map->peers[id].region is valid and if it isnt then go to the out to map to region.

Reply:
There is a check in `smr_map_to_region` that skips initialization if it's already been set so I don't think it should do a double mapping. The addition of the mapping was added because if the mapping failed on av_insert, we need to trigger it again on the next map_add (in progress_connreq). Not saying it isn't related but it should be skipping the duplicate mapping in the case you're talking about